ui/nuxeo-document-layout.js

```diff
@@ -0,0 +1,305 @@
+        </div>
+        <nuxeo-layout
+          id="layout"
+          href="[[_href]]"
+          model="[[_model]]"
+          error="[[i18n('documentLayout.notFound', layout, document.type)]]"
+          on-element-changed="_elementChanged"
+        >
+        </nuxeo-layout>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-layout';
+    }
+
+    static get importMeta() {
+      return import.meta;
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The document for which the layout will be stamped.
+         */
+        document: {
+          type: Object,
+          notify: true,
+        },
+        /**
+         * An id denoting the name of the layout to be stamped.
+         */
+        layout: {
+          type: String,
+          value: 'view',
+          reflectToAttribute: true,
+        },
+        /**
+         * The template to be used to build the layout href.
+         */
+        hrefTemplate: {
+          type: String,
+          // eslint-disable-next-line no-template-curly-in-string
+          value: () => '${document.type}/nuxeo-${document.type}-${layout}-layout.html',
+        },
+        /**
+         * The base url for the layout href.
+         */
+        hrefBase: {
+          type: String,
+          value: '',
+        },
+        /**
+         * The function used to build the layout href, using `hrefTemplate` as the function body.
+         */
+        hrefFunction: {
+          type: Function,
+          computed: '_buildHrefFn(hrefTemplate)',
+        },
+        _model: {
+          type: Object,
+          value: {},
+          readOnly: true,
+        },
+        _href: {
+          type: String,
+          readOnly: true,
+        },
+        _errorMessages: {
+          type: Array,
+          readOnly: true,
+          value: [],
+        },
+      };
+    }
+
+    static get observers() {
+      return ['_loadLayout(document, layout, hrefFunction, hrefBase)'];
+    }
+
+    /**
+     * Fired on the next render after the document layout changes.
+     *
+     * @event document-layout-changed
+     * @param {string} layout The document layout.
+     * @param {object} element The document layout element.
+     */
+
+    /**
+     * Returns the layout element.
+     */
+    get element() {
+      return this.$.layout.element;
+    }
+
+    /**
+     * Runs form validation on the document layout.
+     */
+    validate() {
+      return this.$.layout.validate();
+    }
+
+    /**
+     * Applies focus on the first layout child with the `autofocus` attribute.
+     *
+     * Note: This method won't do anything if there's already a child element focused.
+     */
+    applyAutoFocus() {
+      const focusableElement = this._getFocusableElement(this.element);
+      if (focusableElement) {
+        focusableElement.focus();
+      }
+    }
+
+    /**
+     * Displays any errors present in a validation report, and invalidates layout widgets bound to document
```

Comment:
nit: `a validation report, and`. I would remove the comma

Reply:
It's an Oxford comma. :)